### PR TITLE
refactor: attribute mapping

### PR DESCRIPTION
**Pull request by hustcc**
https://github.com/antvis/G2Plot/issues/1540

 - [x] 重构底层视觉通道的映射逻辑，让所有图形的视觉通道映射保持一致

各个图全部 review 一遍，跑通单测，demo 

 - [x] 折
 - [x] 面积
 - [x] 柱
 - [ ] 饼
 - [x] 散点
 - [x] 双轴
 - [x] 雷达
 - [ ] ...

### CODE REVIEW

File: src/adaptor/geometries/base.ts [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+   * g.shape('x*y', (x, y) => 'rect');
+   * g.shape('shape*x*y', (shape, x, y) => 'rect');
+   */
+  if (isString(shape)) {
+    geometry.shape(shape);
+  } else if (isFunction(shape)) {
+    const mappingFields = getMappingField(options, 'shape');
+    geometry.shape(mappingFields.join('*'), getMappingFunction(mappingFields, shape));
+  } else if (shapeField) {
+    geometry.shape(shapeField, shape ? shape : undefined);
```
Comment (lxfu1) [inline src/adaptor/geometries/base.ts:130]: - geometry.shape(shapeField, shape ? shape : undefined);
- geometry.shape(shapeField );
只有shapeField时两者最终效果一样吗？

Comment (hustcc) [reply src/adaptor/geometries/base.ts:130]: 应该是这样：geometry.shape(shapeField, isArray(shape) ? shape : undefined);

Comment (lxfu1) [reply src/adaptor/geometries/base.ts]: 我担心两者 geometry.shape(shapeField )  geometry.shape(shapeField,  undefined) 结果不一样。

File: src/adaptor/geometries/base.ts [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+  } else if (colorField) {
+    geometry.color(colorField, isArray(color) ? color : undefined);
+  }
+
+  /**
+   * shape 的几种情况
+   * g.shape('rect');
+   * g.shape('shape', ['rect', 'circle']);
+   * g.shape('x*y', (x, y) => 'rect');
+   * g.shape('shape*x*y', (shape, x, y) => 'rect');
```
Comment (Me-Momo) [inline src/adaptor/geometries/base.ts:122]: `{shape, x, y }`

File: src/adaptor/geometries/base.ts [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+
+  // 创建 geometry
+  const geometry = chart[type](args).position(`${xField}*${yField}`);
+
+  /**
+   * color 的几种情况
+   * g.color('red');
+   * g.color('color', ['red', 'blue']);
+   * g.color('x*y', (x, y) => 'red');
+   * g.color('color*x*y', (color, x, y) => 'red');
```
Comment (Me-Momo) [inline src/adaptor/geometries/base.ts:106]: `{ color, x, y) => 'red'); } `

**Review by Me-Momo — APPROVED** [review]

**Pull request merged**